src/explore-education-statistics-admin/src/components/ProtectedRoute.tsx

```diff
@@ -13,24 +13,31 @@ interface ProtectedRouteProps extends RouteProps {
 
 const basicAccessCheck = (user: User) => user.permissions.canAccessSystem;
 
-const AuthenticationCheckingComponent = ({
+/**
+ * Creates a <Route> that firstly checks the user's authentication
+ * status and then renders the protected component if the user has been
+ * successfully authorized, redirects the user to the sign-in page
+ * if in need of authentication, or renders a Forbidden page if not
+ * authorized.
+ */
+const ProtectedRoute = ({
   component,
   allowAnonymousUsers = false,
   protectionAction,
-  ...props
+  ...rest
 }: ProtectedRouteProps) => {
   const { user } = useAuthContext();
 
-  const { handleManualErrors } = useContext(ErrorControlContext);
+  const { handleManualErrors } = useErrorControl();
 
-  let protectedByAction = false;
+  let accessDenied = false;
 
   if (user) {
     const accessCheck = protectionAction || basicAccessCheck;
-    protectedByAction = !accessCheck(user);
+    accessDenied = !accessCheck(user);
   } else {
     const denyAccessToNonLoggedInUsers = !allowAnonymousUsers;
-    protectedByAction = denyAccessToNonLoggedInUsers;
+    accessDenied = denyAccessToNonLoggedInUsers;
   }
 
   if (!component) {
```

Comment:
Why bother checking if there is a component or not? This should be handled internally by React Router.

Reply:
Happy for React Router to do this - was probably being oversafe.